### PR TITLE
Adds random.choices to the random module

### LINKED ISSUE #1188: `random.choices` isn't supported yet
Looking through [`random.js`](https://github.com/skulpt/skulpt/blob/master/src/lib/random.js), it seems that `random.choices` (plural) isn't supported yet. Happy to work on a PR that implements it.

### PR DESCRIPTION
**Pull request by Duncanr-glitch**
Fixes #1188 

Copied the choices method from cpyhon's random module and then put it into a new random.py file that calls all the random.js functions renamed as _random.js. 

Only works in python3

### CODE REVIEW

**Review by bnmnetp — APPROVED** [review]

File: test/unit3/test_random.py [hunk 1]
```diff
@@ -108,6 +109,365 @@ def test_sample_all(self):
...
+        with self.assertRaises(ValueError):
+            random.choices('AB', [0.0, float('inf')])
+        with self.assertRaises(ValueError):
+            random.choices('AB', [-float('inf'), 123])
+        with self.assertRaises(ValueError):
+            random.choices('AB', [0.0, float('nan')])
+        with self.assertRaises(ValueError):
+            random.choices('AB', [float('-inf'), float('inf')])
+class TestBasicOps:
+    # Superclass with tests common to all generators.
```
Comment (s-cork) [inline test/unit3/test_random.py:218]: Thanks for adding these
can we move TestBasicOps to the top of the file to better match the CPython test file
(makes future edits easier)
(and similar for MersenneTwister_TestBasicOps)

**Review by s-cork — CHANGES_REQUESTED** [review]
Thanks @Duncanr-glitch, a couple of suggestions then I think it's good to go.

**Comment by Duncanr-glitch** [comment]
All of that should be done I think :) Thanks for the feedback!

**Review by s-cork — APPROVED** [review]
LGTM - i'm happy with commenting out the python 2 syntax error for keyword only arguments.
(it doesn't break existing python 2 code)

I don't think @bnmnetp uses python 2 anymore, but @meredydd might want to just confirm that this change is ok.

**Comment by bnmnetp** [comment]
python3 only

File: src/lib/random.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+from _random import *
+from math import floor as _floor, isfinite as _isfinite
+from itertools import accumulate as _accumulate, repeat as _repeat
+from bisect import bisect as _bisect
+from sys import version_info
+
+if 3 in version_info:
+    exec("""def choices(population, weights=None, *, cum_weights=None, k=1):
```
[the remark this answers is not in the document]
Comment (meredydd) [reply src/lib/random.py]: Agreed, yes.

**Comment by meredydd** [comment]
(Apologies for the long delay here!)

Yes, I am happy with a change that doesn't break existing python 2 code but does accidentally expose some python 3 features in py2 mode (this is a choice we make with the stdlib all the time, and tbh even Anvil will probably want to start deprecating our py2 support soon)

**Pull request merged**